web/impact/impact/v1/views/user_program_confirmation_detail_view.py

```diff
@@ -0,0 +1,107 @@
+from django.contrib.auth import get_user_model
+from django.db.models import F
+from django.template import loader
+from rest_framework.exceptions import ParseError
+from rest_framework.response import Response
+from accelerator.models import (
+    UserRole,
+    Program,
+    ProgramRoleGrant,
+    ProgramRole
+)
+from impact.minimal_email_handler import MinimalEmailHandler
+from impact.permissions.v1_api_permissions import UserDetailViewPermission
+from impact.v1.helpers import UserProgramConfirmationHelper
+from impact.v1.views.user_detail_view import (
+    BaseDetailView,
+    NO_USER_ERROR
+)
+
+User = get_user_model()
+
+SUBJECT = '[MassChallenge] Confirmed: Your Mentorship Commitment'
+INVALID_INPUT_ERROR = "'{}' must be a list of numbers."
+
+
+def extract_values(request, key):
+    try:
+        return [int(i) for i in request.data.get(key, [])]
+    except ValueError:
+        raise ParseError(INVALID_INPUT_ERROR.format(key))
+
+
+class UserProgramConfirmationDetailView(BaseDetailView):
```

Comment:
Since this does not return details about an object in the database, I don't think the "Detail" part of this name is correct. What this does is it allows mentors to confirm their participation in a program, so maybe it should be called `MentorParticipationConfirmView`?